pkg/minio/api-handlers.go

```diff
@@ -461,6 +461,9 @@ func (h objectAPIHandlersWrapper) HeadBucketHandler(w http.ResponseWriter, r *ht
 	if err := h.bucketPrefixSubstitution(w, r, "HeadBucket"); err != nil {
 		return
 	}
+	if !h.checkBucketExistence(r) {
+		h.core.PutBucketHandler(w, r)
```

Comment:
Head bucket shouldn't creare a bucket if it doesn't exists
Read the reference: https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadBucket.html